ckan/plugins/interfaces.py

```diff
@@ -200,6 +202,42 @@ def notify_after_commit(self, entity, operation):
         pass
 
 
+class IFeed(Interface):
+    """
+    Allows extension of the default Atom feeds
+    """
+
+    def get_feed_class(self):
+        """
+        Allows plugins to provide a custom class to generate feed items.
+
+        The feed item generator class should accept the following parameters
+        on the constructor method:
```

Comment:
I guess not all of these are mandatory. Can you indicate which are (if any) and if not mention that they are all optional.